File: host_gui/gui/cursor.py
from PySide6.QtCore import Qt, QLineF, QPointF
from PySide6.QtGui import QColor, QPainter, QPen
from PySide6.QtWidgets import QWidget

class Cursor(QWidget):
    def __init__(self, parent):
        super().__init__(parent)
        self.setAttribute(Qt.WA_TransparentForMouseEvents)
        self.setAttribute(Qt.WA_NoSystemBackground)
        self.setVisible(False)

        self.target = QPointF(0, 0)

    def paintEvent(self, event):
        painter = QPainter(self)
        pen = QPen()
        pen.setDashPattern([10, 10])
        pen.setCapStyle(Qt.FlatCap)
        pen.setColor(QColor.fromHslF(0, 0, 0.5, 0.5))
        painter.setPen(pen)

        # Horizontal
        painter.drawLine(QLineF(0, self.target.y(),
                                self.width(), self.target.y()))
        # Vertical
        painter.drawLine(QLineF(self.target.x(), 0,
                                self.target.x(), self.height()))

File: host_gui/gui/cursor.cc
#include "cursor.h"

#include <QPainter>
#include <QPen>

Cursor::Cursor(QWidget* parent) : QWidget(parent) {
  setAttribute(Qt::WA_TransparentForMouseEvents);
  setAttribute(Qt::WA_NoSystemBackground);
  setVisible(false);
}

void Cursor::paintEvent(QPaintEvent* event) {
  QPainter painter(this);

  QPen pen;
  pen.setDashPattern({10, 10});
  pen.setCapStyle(Qt::FlatCap);
  pen.setColor(QColor::fromHslF(0, 0, 0.5, 0.5));
  painter.setPen(pen);

  const QPointF center = target_.center();
  // Horizontal
  painter.drawLine(QLineF(0, center.y(), width(), center.y()));
  // Vertical
  painter.drawLine(QLineF(center.x(), 0, center.x(), height()));

  painter.drawRect(target_);
}
